lang/cs/Org.Apache.REEF.Wake/Remote/Impl/WritableLink.cs

```diff
@@ -174,7 +174,8 @@ public T Read()
                 return default(T);
             }
 
-            T value = _cache.GetInstance(dataType);
+            T value = (T) _injector.ForkInjector().GetInstance(dataType);
+            //T value = _cache.GetInstance(dataType);
 
             if (value == null)
```

Comment:
This should never happen. You can either throw in this case or wait for the null exception to happen.

You could, however, catch `InjectionException` from the `GetInstance` call and use `default(T)` in those cases.